internal/server/GameLoop.go

```diff
@@ -49,6 +49,24 @@ func (s *Server) RunBikeSwitch() {
 	s.GetLeavingDecisions()
 	// process joining requests from last round
 	s.ProcessJoiningRequests()
+	//process the kickout request
+	s.HandleKickoutProcess(objects.GetMegaBike())
```

Comment:
should be done before process joining requests (so that it is known exactly how many empty seats there are for new people to join)